analyzers/src/SonarAnalyzer.CSharp/Rules/MultilineBlocksWithoutBrace.cs

```diff
@@ -28,143 +28,118 @@
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckLoop(c, ((ForStatementSyntax)c.Node).Statement),
-                SyntaxKind.ForStatement);
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckLoop(c, ((ForStatementSyntax)c.Node).Statement),
+            SyntaxKind.ForStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckLoop(c, ((ForEachStatementSyntax)c.Node).Statement),
-                SyntaxKind.ForEachStatement);
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckLoop(c, ((ForEachStatementSyntax)c.Node).Statement),
+            SyntaxKind.ForEachStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => CheckIf(c, (IfStatementSyntax)c.Node),
-                SyntaxKind.IfStatement);
-        }
+        context.RegisterSyntaxNodeActionInNonGenerated(
+            c => CheckIf(c, (IfStatementSyntax)c.Node),
+            SyntaxKind.IfStatement);
+    }
 
-        private static void CheckLoop(SyntaxNodeAnalysisContext context, StatementSyntax statement)
+    private static void CheckLoop(SyntaxNodeAnalysisContext context, StatementSyntax statement)
+    {
+        if (!IsNestedStatement(statement))
         {
-            if (IsNestedStatement(statement))
-            {
-                return;
-            }
-
             CheckStatement(context, statement, "in a loop", "only once");
         }
+    }
 
-        private static void CheckIf(SyntaxNodeAnalysisContext context, IfStatementSyntax ifStatement)
+    private static void CheckIf(SyntaxNodeAnalysisContext context, IfStatementSyntax ifStatement)
+    {
+        if (!ifStatement.GetPrecedingIfsInConditionChain().Any()
+            && !IsNestedStatement(ifStatement.Statement)
+            && GetLastStatementInIfChain(ifStatement) is { } lastStatementInIfChain
+            && !IsStatementCandidateLoop(lastStatementInIfChain))
         {
-            if (ifStatement.GetPrecedingIfsInConditionChain().Any())
-            {
-                return;
-            }
-
-            if (IsNestedStatement(ifStatement.Statement))
-            {
-                return;
-            }
-
-            var lastStatementInIfChain = GetLastStatementInIfChain(ifStatement);
-            if (IsStatementCandidateLoop(lastStatementInIfChain))
-            {
-                return;
-            }
-
             CheckStatement(context, lastStatementInIfChain, "conditionally", "unconditionally");
         }
+    }
 
-        private static bool IsNestedStatement(StatementSyntax nested) =>
-            nested.IsKind(SyntaxKind.IfStatement) ||
-            nested.IsKind(SyntaxKind.ForStatement) ||
-            nested.IsKind(SyntaxKind.ForEachStatement) ||
-            nested.IsKind(SyntaxKind.WhileStatement);
+    private static StatementSyntax GetLastStatementInIfChain(IfStatementSyntax ifStatement)
```

Comment:
Unrelated improvement - remove unneeded `Get` prefix
```suggestion
    private static StatementSyntax LastStatementInIfChain(IfStatementSyntax ifStatement)
```